lib/iex/lib/iex/info.ex

```diff
@@ -35,13 +35,27 @@ defimpl IEx.Info, for: Atom do
       end
 
     mod_info = mod.module_info()
-    ["Module bytecode": module_object_file(mod),
-     "Source": module_source_file(mod_info),
-     "Version": module_version(mod_info),
-     "Compile options": module_compile_options(mod_info),
-     "Description": "#{extra}Call #{inspect mod}.module_info() to access metadata.",
-     "Raw representation": ":" <> inspect(Atom.to_string(mod)),
-     "Reference modules": "Module, Atom"]
+    generic_info =
+      ["Module bytecode": module_object_file(mod),
+       "Source": module_source_file(mod_info),
+       "Version": module_version(mod_info),
+       "Compile options": module_compile_options(mod_info),
+       "Description": "#{extra}Call #{inspect mod}.module_info() to access metadata."]
+
+    specific_info =
+      if _protocol? = function_exported?(mod, :__protocol__, 1) do
```

Comment:
`_protocol?` can definitely go, I think it was a residue of an older version of this code that I forgot to kill :) As for the loading, first of all yes it's loaded since we only end up in `info_module/1` if `Code.ensure_loaded?/1` returns true in the `cond` a few lines above. And anyways, `function_exported?/3` should not load the module :)